### DISCUSSION TITLE
Using NWB for auditory neurophysiology

### DISCUSSION THREAD

**Discussion by aheeringa** [dandi/helpdesk#130] [General]
Dear NWB community, 

I would like to convert my extracellularly recorded data from single auditory nerve fibers to the NWB format. However, I am running into some structuring problems with the ElectricalSeries in NWB. 

My data is organized as follows: 
In a given animal, I record with a glass electrode from one auditory nerve fiber at a time, eventually I end up with recordings from ~20 fibers in 1 animal. 
From each fiber, I record responses to a number of recording schemes. In one recording scheme, e.g. determining its best frequency response, I present a number of tone frequencies (e.g. 20 different frequencies), which are each repeated N times and presented randomly within each repetition.
So I end up with ~100 - 600 trials per recording, 4-5 different recordings from one fiber, and ~20 fibers per animal. 
In the metadata, it should be clear which trials belong together, as well as which recordings derived from the same fiber. Also the metadata should describe for each trial the acoustic characteristics. 

In the NWB format, extracellular data is organized as one long trial, with one starting time. This is not a convenient format for my data as I would need one NWB file per trial and it would not be clear which files belong to which recording / fiber / animal. The intracellular data may be more convenient, although then it would not be clear that my data derives from an extracellularly recorded signal.  
What would be your suggestion for the conversion? How can I best organize these data into the NWB format?

Thank you for you advice, 

Kind regards,
Amarins Heeringa
Carl von Ossietzky University Oldenburg

**Comment by CodyCBakerPhD** [comment]
> This is not a convenient format for my data as I would need one NWB file per trial and it would not be clear which files belong to which recording / fiber / animal.

To clarify, it sounds like you only perform the recording during trials, not continuously?

If you know the starting time of each trial (and each recording for that trial) relative to the global start time, you can also use a `timestamps` (vector, length of the number of frames) for a single `ElectricalSeries` that is collected in an irregular fashion - this would indicate the gaps in recording between trials (which would be a full table, one row per trial, one table per file)

**Reply by aheeringa** [reply]
Yes, I only record during the trials. Thanks for the option, I will look into this! Is there a way to indicate then which files all derive from the same fiber? Because these metadata are crucial to interpret the recordings as well.

**Reply by CodyCBakerPhD** [reply]
You're not including all fibers in the same file?

Otherwise, I don't see anything described above that would violate the 'one session per NWB file' convention, with the file per session containing all recordings from all trials from all fibers
